File: clients/javascript/src/models/publishIn.ts
// this file is @generated
import {
    type MsgIn,
    MsgInSerializer,
} from './msgIn';





export interface PublishIn {
    msgs: MsgIn[];
topic: string;
}

export const PublishInSerializer = {
    _fromJsonObject(object: any): PublishIn {
        return {
            msgs: object['msgs'].map((item: MsgIn) => MsgInSerializer._fromJsonObject(item)),
            topic: object['topic'],
            };
    },

    _toJsonObject(self: PublishIn): any {
        return {
            'msgs': self.msgs.map((item) => MsgInSerializer._toJsonObject(item)),
            'topic': self.topic,
            };
    }
}

File: clients/rust/src/models/publish_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::msg_in::MsgIn;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct PublishIn {
    pub msgs: Vec<MsgIn>,

    pub topic: String,
}

impl PublishIn {
    pub fn new(msgs: Vec<MsgIn>, topic: String) -> Self {
        Self { msgs, topic }
    }
}
